### ISSUE TITLE
Fail to create a pmml file 

### ISSUE THREAD

**Issue by sharonzagh** [jpmml/sklearn2pmml#250]
Hi, 
I am trying to create pmml file using the code below. I am able to fit the pipeline but not to create the pmml file 

```python
def missing_categoric(replace='missing'):
    return CategoricalDomain(
        missing_value_treatment="as_value",
        missing_value_replacement=replace,
        with_statistics=False,
        with_data=False,
    )

def add_missing_categoric(mapper, features: list, replacement):
    feature_defs = [
        (col_name, missing_categoric(replacement)) for col_name in features
    ]
    mapper.features = mapper.features + feature_defs

def missing_numeric(replace=-999.01):
    return ContinuousDomain(
        missing_value_treatment="as_value",
        missing_value_replacement=replace,
        with_statistics=False,
        with_data=False,
    )

def add_missing_numeric(mapper, features: list, replacement):
    feature_defs = [
        (col_name, missing_numeric(replacement)) for col_name in features
    ]
    mapper.features = mapper.features + feature_defs

def numerical_categorical_transform(mapper: DataFrameMapper, features: list,replace=-999.01):
    feature_defs = [
        (col,[ContinuousDomain(missing_value_treatment="as_value",missing_value_replacement=replace,with_statistics=False,with_data=False,),
              CastTransformer(float)]) 
        for col in features
    ]
    mapper.features = mapper.features + feature_defs

def categorical_str_transform(mapper: DataFrameMapper, features: list,replace='missing'):
    feature_defs = [
    (col,[CategoricalDomain(missing_value_treatment="as_value",
                                   missing_value_replacement=replace,
                                   with_statistics=False,
                                   with_data=False,
                                  ),
          Alias(StringNormalizer(function = "lowercase"),name=col)]
    ) 
        for col in features
    ]
    mapper.features = mapper.features + feature_defs

def add_expression2(mapper, input_fields: list, expression: str, alias: str):
    feature_defs =  [
        (col , [Alias(ExpressionTransformer(expression), name=alias), 
                CastTransformer(float)
               ],
        {'alias': alias}
    )
        for col in input_fields
    ]
    mapper.features = mapper.features + feature_defs


def create_map_df_lift(table):
    df = pd.DataFrame(columns = ['feature', 'category','new_feature_name', 'value'])
    
    for feature in table['feature'].unique().tolist():
  
        for index, row in table.loc[table['feature'] == feature].iterrows():
            category = row['category']
            lift = row['lift']
            new_feature_name = 'LR_'+feature
            
            df = df.append({'feature' : feature, 'category' : category, 'new_feature_name': new_feature_name, 'value': lift} , ignore_index=True)
            
    return(df)

def add_encoding_to_recorder(mapper, encoding_df):
    ''' function that creates features in recorder from enncoding df '''
    for feature in encoding_df['feature'].unique().tolist():
        print(feature)
        str_expression = ''
        for index, row in encoding_df.loc[encoding_df['feature'] == feature].iterrows():

            category = row['category']

            value = row['value']

            new_feature_name = row['new_feature_name']

            str_expression = str_expression+ f" {value} if X == '{category}' else "
    
        
        str_expression = str_expression + "X"
        add_expression2(mapper,[feature], str_expression,new_feature_name)
    

def calculate_lift_features(lift_table,recorder_obj):
    encoding_df_lift = create_map_df_lift(lift_table)
    add_encoding_to_recorder(recorder_obj, encoding_df_lift)  
    return

recorder= DataFrameMapper([])
recorder.features = []

numerical_categorical_transform(recorder, numerical_categorical)
add_missing_categoric(recorder, numerical_categorical_with_missing, 'missing')
categorical_str_transform(recorder,categorical_str_features)
add_missing_numeric(recorder, col_amount, 0)

calculate_lift_features(lift_table, recorder)


recorder.df_out = True
res = recorder.fit_transform(df.reset_index(),y_train)

list_cat_miss = res.select_dtypes(include='object').columns.tolist()

pipeline = PMMLPipeline([
    ("mapper", recorder),
    #("select", ColumnTransformer([("drop", "drop", list_cat_miss)], remainder='passthrough')),
    ("select", ColumnTransformer([("passthrough", "passthrough", list_not_cat_miss)], remainder='drop')),
    ("classifier", XGBClassifier())
])

pipeline.fit(rdf,y_train)

sklearn2pmml(pipeline, "pipeline.pmml")
```

This is the error message I get: 
RuntimeError: The JPMML-SkLearn conversion application has failed. The Java executable should have printed more information about the failure into its standard output and/or standard error streams``

**Comment by vruusmann** [comment]
> The Java executable should have printed more information about the failure into its standard output and/or standard error streams

Tell me what was printed to standard output and/or standard error streams. That's the signal I'm looking for.

**Comment by sharonzagh** [comment]
Standard output is empty
Standard error:
Oct 11, 2020 2:25:38 PM org.jpmml.sklearn.Main run
INFO: Parsing PKL..
Oct 11, 2020 2:25:38 PM org.jpmml.sklearn.Main run
INFO: Parsed PKL in 47 ms.
Oct 11, 2020 2:25:38 PM org.jpmml.sklearn.Main run
INFO: Converting..
Oct 11, 2020 2:25:38 PM org.jpmml.sklearn.Main run
SEVERE: Failed to convert
java.lang.IllegalArgumentException: builtins.float
	at numpy.DType.getDataType(DType.java:89)
	at sklearn2pmml.preprocessing.CastTransformer.encodeFeatures(CastTransformer.java:48)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn_pandas.DataFrameMapper.initializeFeatures(DataFrameMapper.java:73)
	at sklearn.Initializer.encodeFeatures(Initializer.java:41)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn.pipeline.Pipeline.encodeFeatures(Pipeline.java:83)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:203)
	at org.jpmml.sklearn.Main.run(Main.java:145)
	at org.jpmml.sklearn.Main.main(Main.java:94)

Exception in thread "main" java.lang.IllegalArgumentException: builtins.float
	at numpy.DType.getDataType(DType.java:89)
	at sklearn2pmml.preprocessing.CastTransformer.encodeFeatures(CastTransformer.java:48)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn_pandas.DataFrameMapper.initializeFeatures(DataFrameMapper.java:73)
	at sklearn.Initializer.encodeFeatures(Initializer.java:41)
	at sklearn.Transformer.updateAndEncodeFeatures(Transformer.java:85)
	at sklearn.pipeline.Pipeline.encodeFeatures(Pipeline.java:83)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:203)
	at org.jpmml.sklearn.Main.run(Main.java:145)
	at org.jpmml.sklearn.Main.main(Main.java:94)

**Comment by vruusmann** [comment]
> java.lang.IllegalArgumentException: builtins.float
> at numpy.DType.getDataType(DType.java:89)
> at sklearn2pmml.preprocessing.CastTransformer.encodeFeatures(CastTransformer.java:48)

It means that your version of SkLearn2PMML package does not think that `builtins.float` is a valid `CastTransformer.dtype` argument.

This functionality has been implemented for quite some time:
https://github.com/jpmml/jpmml-python/blob/1.0.6/src/main/java/numpy/DType.java#L68-L69

The mismatch between Java source code line numbers (this IllegalArgumentException should be thrown on line 93, but according to your report is thrown in line 89) suggests that you're simply using an outdated SkLearn2PMML package version:
https://github.com/jpmml/jpmml-python/blob/1.0.6/src/main/java/numpy/DType.java#L92-L93

TLDR: Please update your SkLearn2PMML package version, and this issue will be resolved.
